comfyui-truss/model/model.py

```diff
@@ -0,0 +1,203 @@
+import os
+import subprocess
+import sys
+import websocket
+import uuid
+import json
+import random
+from io import BytesIO
+import base64
+import urllib.request
+import urllib.parse
+from typing import Dict
+from multiprocessing import Process
+import requests
+import tempfile
+
+side_process = None
+original_working_directory = os.getcwd()
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self._data_dir = kwargs["data_dir"]
+        self._model = None
+        self.ws = None
+        self.server_address = "127.0.0.1:8188"
+        self.client_id = str(uuid.uuid4())
+
+
+        def download_model(self, model_url, destination_path):
+            print(f"Downloading model {model_url} ...")
+            try:
+                response = requests.get(model_url, stream=True)
+                response.raise_for_status()
+
+                # Open the destination file and write the content in chunks
+                with open(destination_path, 'wb') as file:
+                    for chunk in response.iter_content(chunk_size=8192):
+                        if chunk:
+                            file.write(chunk)
+
+                print(f"Downloaded file to: {destination_path}")
+            except requests.exceptions.RequestException as e:
+                print(f"Download failed: {e}")
+
+    def download_tempfile(self, file_url, filename):
```

Comment:
can you move these helper functions to a `helpers.py` file so that the model class is more readable?